lib/hanami/model.rb

```diff
@@ -1,172 +1,84 @@
-    # original. It will inherit the configuration, but all the changes that
-    # happen after the duplication, won't be reflected on the other copies.
-    #
-    # @return [Module] a copy of Hanami::Model
-    #
-    # @since 0.2.0
+    # @since x.x.x
     # @api private
-    #
-    # @example Basic usage
-    #   require 'hanami/model'
-    #
-    #   module MyApp
-    #     Model = Hanami::Model.dupe
-    #   end
-    #
-    #   MyApp::Model == Hanami::Model # => false
-    #
-    #   MyApp::Model.configuration ==
-    #     Hanami::Model.configuration # => false
-    #
-    # @example Inheriting configuration
-    #   require 'hanami/model'
-    #
-    #   Hanami::Model.configure do
-    #     adapter type: :sql, uri: 'sqlite3://uri'
-    #   end
-    #
-    #   module MyApp
-    #     Model = Hanami::Model.dupe
-    #   end
-    #
-    #   module MyApi
-    #     Model = Hanami::Model.dupe
-    #     Model.configure do
-    #       adapter type: :sql, uri: 'postgresql://uri'
-    #     end
-    #   end
-    #
-    #   Hanami::Model.configuration.adapter_config.uri # => 'sqlite3://uri'
-    #   MyApp::Model.configuration.adapter_config.uri # => 'sqlite3://uri'
-    #   MyApi::Model.configuration.adapter_config.uri # => 'postgresql://uri'
-    def self.dupe
-      dup.tap do |duplicated|
-        duplicated.configuration = Configuration.new
-      end
+    def self.container
+      raise 'Not loaded' unless loaded?
+      @container
     end
 
-    # Duplicate the framework and generate modules for the target application
-    #
-    # @param mod [Module] the Ruby namespace of the application
-    # @param blk [Proc] an optional block to configure the framework
-    #
-    # @return [Module] a copy of Hanami::Model
-    #
-    # @since 0.2.0
-    #
-    # @see Hanami::Model#dupe
-    # @see Hanami::Model::Configuration
-    #
-    # @example Basic usage
-    #   require 'hanami/model'
-    #
-    #   module MyApp
-    #     Model = Hanami::Model.dupe
-    #   end
-    #
-    #   # It will:
-    #   #
-    #   # 1. Generate MyApp::Model
-    #   # 2. Generate MyApp::Entity
-    #   # 3. Generate MyApp::Repository
-    #
-    #   MyApp::Model      == Hanami::Model # => false
-    #   MyApp::Repository == Hanami::Repository # => false
-    #
-    # @example Block usage
-    #   require 'hanami/model'
-    #
-    #   module MyApp
-    #     Model = Hanami::Model.duplicate(self) do
-    #       adapter type: :memory, uri: 'memory://localhost'
-    #     end
-    #   end
-    #
-    #   Hanami::Model.configuration.adapter_config # => nil
-    #   MyApp::Model.configuration.adapter_config # => #<Hanami::Model::Config::Adapter:0x007ff0ff0244f8 @type=:memory, @uri="memory://localhost", @class_name="MemoryAdapter">
-    def self.duplicate(mod, &blk)
-      dupe.tap do |duplicated|
-        mod.module_eval %{
-          Entity = Hanami::Entity.dup
-          Repository = Hanami::Repository.dup
-        }
```

Comment:
In order for Hanami to work with Hanami::Model, the [Hanami loader](https://github.com/hanami/hanami/blob/master/lib/hanami/loader.rb#L134) will have to be updated to call `Hanami::Model::load!` instead of `Hanami::Model::duplicate`.